pages/api/roles/index.ts

```diff
@@ -68,12 +69,35 @@ function handler(req: NextApiRequest, res: NextApiResponse) {
   }
 }
 
-// TODO: Optional Search query. Will be done in another PR to unblock others.
-
 async function handleGet(req: NextApiRequest, res: NextApiResponse<ApiResponse<RoleListData[]>>) {
+  const { companyId, searchQuery } = req.query;
+
+  const queryCompanyId = isInteger(companyId as string) ? Number(companyId) : undefined;
+
+  const searchWords = typeof searchQuery === 'string' ? (searchQuery as string).split(' ') : [];
```

Comment:
As for checking the type of `searchQuery` to make sure that it is of type `string` and not `string[]`, I think that we can leave it out, since we can assume that we are the ones specifying the query params from the frontend and we will always pass a string. I don't mind if you add it back in though.

Reply:
Would still prefer to keep the typeof since we don't know that it is a string.
```suggestion
async function handleGet(req: NextApiRequest, res: NextApiResponse<ApiResponse<RoleListData[]>>) {
  const { companyId, searchQuery } = req.query;

  const queryCompanyId = isInteger(companyId as string) ? Number(companyId) : undefined;

  const searchWords = typeof searchQuery === 'string' ? (searchQuery as string).split(' ').filter(x => !isEmpty(x) : [];
```